### DISCUSSION TITLE
Does it support Twitter's webhooks and activity API?

### DISCUSSION THREAD

**Discussion by nnx0r** [plhery/node-twitter-api-v2#204] [Q&A] [answered]
I tried to register webhook simply by using:

```
// Register Webhooks
const webhook = await client.v1.post(
  'account_activity/all/dev/webhooks?url=http://46aa-178-149-218-192.ngrok.io/webhook/twitter',
)
```

Request fails. I tried with `await client.v2.post` as well. 

I might not know how to do this right... Wonder if it's bundled within the lib, but I can't find it within examples.

Useful links:
https://developer.twitter.com/en/docs/twitter-api/premium/account-activity-api/guides/managing-webhooks-and-subscriptions
https://developer.twitter.com/en/docs/twitter-api/premium/account-activity-api/guides/account-activity-data-objects

**Comment by alkihis** [comment, marked as answer]
Hi,

No, Account Activity API is not wrapped by specific methods.
You are making 2 little mistakes in your request:

```ts
// use .v1, because API prefix is https://api.twitter.com/1.1
// >>> Don't forget to add the .json at the end of the URL !
// >>> Give parameters in body, not in URL!
const webhook = await client.v1.post('account_activity/all/dev/webhooks.json', { url: 'http://46aa-178-149-218-192.ngrok.io/webhook/twitter' })
```

**Reply by nnx0r** [reply]
Ouch, I see... Thanks for heads-up. It's more complex than this, to register webhook and subscribe to events. I spent some time attempting this. Twitter docs are not greatest IMO :/ 

Thanks a lot for making this. It's the best api for Node!

**Comment by nnx0r** [comment]
I found an alternative lib for activity events only. It's unfortunate that it's not all covered by a single lib, but this will do. 
Hope to see Activity Api implemented in this package soon. 

Thanks @alkihis and keep up the good work!

**Reply by alkihis** [reply]
I will be happy to know which lib you've used for Account Activity API ! :)
I develop this lib on my free time, so implementing AAAPI endpoints is planned, but not now.
But it isn't planned to implement support of webhooks (that requires a webserver) ; only REST endpoints.

**Reply by nnx0r** [reply]
I used `twict`. This one has endpoitns to register, delete and subcribe hooks, but also has an event listener to handle different events (one event of DM being read throws error tho, as it's not handled)

Aside it has the server to always run on specified port and uses NGROK ofc to proxy requests. 
I'm not really a fan that it has express bundled in, but it does the job.

---

I understand. Happy to hear that it is planned. It's really nice library. Wish if it had more usage and perhaps sponsors to keep you and this project going.

**Reply by winstxnhdw** [reply]
@dvlden Were you ever able to get `twict` working without express? I am not too sure how decoupled his code is.

**Reply by nnx0r** [reply]
@winstxnhdw I think it uses express in its Activity Class. So no, I did not get it working without Express, it comes with it out of the box. So I used that. 🤷‍♂️

**Reply by winstxnhdw** [reply]
I ended up forking `twict` and removing Express. If anyone is interested, it's https://github.com/winstxnhdw/webhoot
